src/test/java/com/google/api/codegen/config/ServiceConfigTest.java

```diff
@@ -0,0 +1,87 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.Authentication;
+import com.google.api.AuthenticationRule;
+import com.google.api.OAuthRequirements;
+import com.google.api.Service;
+import com.google.api.codegen.util.ProtoParser;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Model;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class ServiceConfigTest {
+  private static final Model protoModel = Mockito.mock(Model.class);
+  private static final ProtoApiModel apiModel = new ProtoApiModel(protoModel);
+  private static final ProtoParser protoParser = Mockito.mock(ProtoParser.class);
+  private static final Interface libraryService = Mockito.mock(Interface.class);
+  private static final Interface foodService = Mockito.mock(Interface.class);
+
+  private static final Service serviceYaml =
+      Service.newBuilder()
```

Comment:
Please use intermediate variables in cases like this, so the final initialization block looks more readable and has less indentations (here it has 9 levels of indentation, ideally there should be no more than 3-4 levels of indentation in any code block).

Reply:
Done.